### PR TITLE
gh-92118: Add test for traceback when exception is modified by ExitStack.__exit__

### PR DESCRIPTION
**Pull request by iritkatriel**

### CODE REVIEW

File: Lib/test/test_contextlib.py [hunk 1]
```diff
@@ -748,6 +748,39 @@ def test_exit_suppress(self):
...
+        try:
+            with self.exit_stack() as stack:
+                stack.callback(raise_exc, ValueError)
+                1/0
+        except ValueError as e:
+            exc = e
+
+        self.assertIsInstance(exc, ValueError)
+        ve_frames = traceback.extract_tb(exc.__traceback__)
+        self.assertEqual(len(ve_frames), 5)
```
Comment (serhiy-storchaka) [inline Lib/test/test_contextlib.py]: It is redundant, And if it fail, the report will be less informative than failing the test which compares lists.

**Review by serhiy-storchaka — APPROVED** [review]
Is it the same in 3.10?

**Comment by iritkatriel** [comment]
> Is it the same in 3.10?

It is.

**Comment by iritkatriel** [comment]
Hmm. contextlib_async tests use this test and get a different traceback.

**Comment by serhiy-storchaka** [comment]
You can parametrize the test. Set the expected result as an attribute of the class.

**Comment by iritkatriel** [comment]
Thanks. The 3.10 backport will fail because it doesn't have import traceback, but I'll fix it.

**Pull request merged**